guvnor-ala/guvnor-ala-ui/guvnor-ala-ui-client/src/main/java/org/guvnor/ala/ui/client/wizard/source/SourceConfigurationPageView.java

```diff
@@ -0,0 +1,312 @@
+                               branch));
+    }
+
+    @Override
+    public void addProject(String projectName) {
+        projects.add(newOption(projectName,
+                               projectName));
+    }
+
+    @Override
+    public void clearOrganizationUnits() {
+        clear(ous);
+    }
+
+    @Override
+    public void clearRepositories() {
+        clear(repos);
+    }
+
+    @Override
+    public void clearBranches() {
+        clear(branches);
+    }
+
+    @Override
+    public void clearProjects() {
+        clear(projects);
+    }
+
+    @EventHandler("runtime-name")
+    private void onRuntimeChange(@ForEvent("change") final Event event) {
+        presenter.onRuntimeNameChange();
+    }
+
+    @EventHandler("ous")
+    private void onOrganizationalUnitChange(@ForEvent("change") final Event event) {
+        presenter.onOrganizationalUnitChange();
+    }
+
+    @EventHandler("repos")
+    private void onRepositoryChange(@ForEvent("change") final Event event) {
+        presenter.onRepositoryChange();
+    }
+
+    @EventHandler("branches")
+    private void onBranchChange(@ForEvent("change") final Event event) {
+        presenter.onBranchChange();
+    }
+
+    @EventHandler("projects")
+    private void onProjectChange(@ForEvent("change") final Event event) {
+        presenter.onProjectChange();
+    }
+
+    private void enable(boolean enabled) {
+        this.runtimeName.setDisabled(!enabled);
+        this.ous.setDisabled(!enabled);
+        this.repos.setDisabled(!enabled);
+        this.branches.setDisabled(!enabled);
+        this.projects.setDisabled(!enabled);
+    }
+
+    private void resetFormState() {
+        setFormStatus(runtimeForm,
+                      FormStatus.VALID);
+        setFormStatus(ouForm,
+                      FormStatus.VALID);
+        setFormStatus(repoForm,
+                      FormStatus.VALID);
+        setFormStatus(branchForm,
+                      FormStatus.VALID);
+        setFormStatus(projectForm,
+                      FormStatus.VALID);
+    }
+
+    private Option newOption(final String text,
+                             final String value) {
+        final Option option = (Option) Window.getDocument().createElement("option");
+        option.setTextContent(text);
+        option.setValue(value);
+        return option;
+    }
+
+    private HTMLElement defaultOption() {
+        final HTMLElement option = Window.getDocument().createElement("option");
+        option.setAttribute("value",
+                            "");
+        option.setAttribute("disabled",
+                            "");
+        option.setAttribute("selected",
+                            "");
+        option.setTextContent(translationService.getTranslation(SourceConfigurationPageView_SelectOption_placeholder));
+        return option;
+    }
+
+    private void clear(final Select select) {
+        for (int i = 0; i < select.getOptions().getLength() + 1; i++) {
```

Comment:
This looks like 'of by one error' - if items we're removing from it are indexed from 0 then the last will have index `getLength() - 1`, but here you seem to be also removing `getLength()`-th element